src/filesystem-cache.ts

```diff
@@ -64,22 +64,70 @@ export class FilesystemCache {
   }
 
   async clearCache(key: string) {
-    fs.rmdirSync(this.getDir(key), {recursive: true});
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.payloadFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.payloadFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.responseFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.responseFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.requestFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.requestFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(this.getDir(key))) {
+      fs.rmdir(this.getDir(key), () => {
+        console.log(`${key} cache Dir Deleted`);
+      });
+    }
   }
 
   async clearAllCache() {
-    fs.rmdirSync(this.getDir(''), {recursive: true});
+    fs.rmdirSync(this.getDir(''), { recursive: true });
   }
 
   cacheContent(key: string, ctx: Koa.Context) {
     const responseHeaders = ctx.response;
     const responseBody = ctx.body;
     const request = ctx.request;
-
+    // check size of stored cache to see if we are over the max number of allowed entries, and max entries isn't disabled with a value of -1 and remove over quota, removes oldest first
+    if (parseInt(this.config.cacheConfig.cacheMaxEntries) !== -1) {
+      const numCache = fs.readdirSync(this.getDir(''));
```

Comment:
On unix-like operating systems, this might give you an "off by two" error, because it should count `.` and `..`, too

Reply:
Tested on a Ubuntu instance as it appears the count doesn't include ./ & ../ so returns correct count,